libraries/extractor/src/main/java/androidx/media3/extractor/ts/PesReader.java

```diff
@@ -165,6 +165,16 @@ private void setState(int state) {
     bytesRead = 0;
   }
 
+  public boolean canConsumeDummyEndOfInput() {
+    // Pusi only payload to trigger end of sample data is only applicable if
+    // pes does not have a length field and body is being read, another exclusion
```

Comment:
In the comment you say: `pes does not have a length field and body is being read`, but the condition check `payloadSize != C.LENGTH_UNSET` which means length field is set. 

The code is doing opposite to what the comment says, please clarify.

Reply:
Sorry you're right, the comment is right and the code in my last commit (when I added the function canConsumeDummyEndOfInput() is wrong, as from the previous commit the correct check is `!pesReader.hasPacketLength()` which means `payloadSize == C.LENGTH_UNSET`. Thanks for spotting it.

Reply:
I have updated it. Thanks for confirming.